balancer/rls/internal/builder.go

```
@@ -0,0 +1,250 @@
+	}
+
+	if cfg.childPolicy == nil {
+		return nil, fmt.Errorf("rls: childPolicy is invalid in service config {%+v}", string(c))
+	}
+	if cfg.targetField == "" {
+		return nil, fmt.Errorf("rls: childPolicyConfigTargetFieldName field is not set in service config {%+v}", string(c))
+	}
+	cpCfg, err := validateChildPolicyConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &lbConfig{
+		kbMap:                kbMap,
+		lookupService:        parsedTarget,
+		lookupServiceTimeout: lookupServiceTimeout,
+		maxAge:               maxAge,
+		staleAge:             staleAge,
+		cacheSizeBytes:       cacheSizeBytes,
+		rpStrategy:           rpStrategy,
+		cpName:               cfg.childPolicy.Name,
+		cpTargetField:        cfg.targetField,
+		cpConfig:             cpCfg,
+	}, nil
+}
+
+// validateChildPolicyConfig validates the child policy config received in the
+// service config. This makes it possible for us to reject service configs
+// which contain invalid child policy configs which we know will fail for sure.
+//
+// It does the following:
+// * Unmarshals the provided child policy config into a map of string to
+//   json.RawMessage. This allows us to add an entry to the map corresponding
+//   to the targetFieldName that we received in the service config.
+// * Marshals the map back into JSON, finds the config parser associated with
+//   the child policy and asks it to validate the config.
+// * If the validation succeeded, removes the dummy entry from the map and
+//   returns it. If any of the above steps failed, it returns an error.
+func validateChildPolicyConfig(cfg *lbParsedConfig) (map[string]json.RawMessage, error) {
+	var childConfig map[string]json.RawMessage
+	if err := json.Unmarshal(cfg.childPolicy.Config, &childConfig); err != nil {
+		return nil, fmt.Errorf("rls: json unmarshal failed for child policy config {%+v}: %v", cfg.childPolicy.Config, err)
+	}
+	childConfig[cfg.targetField], _ = json.Marshal(dummyChildPolicyTarget)
```

Do you mean to ignore the error here?